otelcol/collector.go

```diff
@@ -150,14 +153,9 @@ func (col *Collector) GetState() State {
 
 // Shutdown shuts down the collector server.
 func (col *Collector) Shutdown() {
-	// Only shutdown if we're in a Running or Starting State else noop
-	state := col.GetState()
-	if state == StateRunning || state == StateStarting {
-		defer func() {
-			_ = recover()
-		}()
```

Comment:
do we care for the recover hook?

Reply:
I assumed it might be to recover from panics during shutdown, so reload can happen anyway.